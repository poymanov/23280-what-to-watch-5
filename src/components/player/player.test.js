import React from "react";
import renderer from "react-test-renderer";
import Player from "./player";

const noop = () => {};

const movie = {
  id: 1,
  name: `test`,
  posterImage: `test`,
  previewImage: `test`,
  backgroundImage: `test`,
  backgroundColor: `test`,
  videoLink: `test`,
  previewVideoLink: `test`,
  description: `test`,
  rating: 1,
  scoresCount: 1,
  director: `test`,
  starring: [`test`],
  runTime: 1,
  genre: `test`,
  released: 1,
  isFavorite: false,
};

describe(`Should Player render correctly`, () => {
  it(`Playing`, () => {
    const tree = renderer
      .create(<Player
        movie={movie}
        isPlaying={true}
        initVideo={noop}
        handleTimeUpdate={noop}
        handleEnded={noop}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Paused`, () => {
    const tree = renderer
      .create(<Player
        movie={movie}
        isPlaying={false}
        initVideo={noop}
        handleTimeUpdate={noop}
        handleEnded={noop}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});


