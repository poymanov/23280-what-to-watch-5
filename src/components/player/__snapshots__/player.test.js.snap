// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Player render correctly Paused 1`] = `
<video
  className="player__video"
  onEnded={[Function]}
  onTimeUpdate={[Function]}
/>
`;

exports[`Should Player render correctly Playing 1`] = `
<video
  className="player__video"
  onEnded={[Function]}
  onTimeUpdate={[Function]}
/>
`;
